src/dataset.rs

```diff
@@ -535,6 +542,106 @@ impl Dataset {
         }
         Ok(Transaction::new(self))
     }
+
+    /// Execute a SQL query against the Dataset. It is equivalent to calling
+    /// [`GDALDatasetExecuteSQL`](https://gdal.org/api/raster_c_api.html#_CPPv421GDALDatasetExecuteSQL12GDALDatasetHPKc12OGRGeometryHPKc).
+    /// Returns a [`sql::ResultSet`], which can be treated just as any other [`Layer`].
+    ///
+    /// Queries such as `ALTER TABLE`, `CREATE INDEX`, etc. have no [`sql::ResultSet`], and return
+    /// `None`, which is distinct from an empty [`sql::ResultSet`].
+    ///
+    /// # Arguments
+    ///
+    /// * `query`: The SQL query
+    /// * `spatial_filter`: Limit results of the query to features that are inside the given
```

Comment:
GDAL doesn't make it clear in the docs I saw, but do you know if "inside the given geometry" means "completely within" or is it only required to intersect?

Reply:
Good point. I don't have any concrete examples, but I do believe it's an intersection, rather than complete containment. For example, if a bbox has a long road go through it, but is not completely contained in it, it still gets returned. I was able to find [this bit](https://gdal.org/user/vector_data_model.html?highlight=spatial%20filter#layer) in the GDAL docs:

> The OGRLayer includes methods for sequential and random reading and writing. Read access (via the `OGRLayer::GetNextFeature()` method) normally reads all features, one at a time sequentially; however, it can be limited to return features **intersecting** a particular geographic region by installing a spatial filter on the OGRLayer (via the `OGRLayer::SetSpatialFilter()` method).

I'll update the language to say "intersecting the given geometry".